ckan/lib/search/common.py

```diff
@@ -65,6 +65,16 @@ def is_available():
 
 def make_connection(decode_dates=True):
     solr_url, solr_user, solr_password = SolrSettings.get()
+
+    if solr_url and solr_user:
+        # Rebuild the URL with the username/password
+        protocol = re.search('(http(?:s)?)://', solr_url).group()
+        solr_url = re.sub(protocol, '', solr_url)
+        solr_url = "{}{}:{}@{}".format(protocol,
+                                       solr_user,
+                                       solr_password,
+                                       solr_url)
```

Comment:
Do the user and password parameters here need to be urlencoded to be included safely in the URL string here?

Reply:
Good catch :+1: